superset/assets/src/dashboard/util/getPublishSubscriberMap.js

```diff
@@ -131,14 +131,38 @@ export default function getPublishSubscriberMap(slices) {
 function updateSlices(slices) {
     let updatedSlices = _.clone(slices);
     updatedSlices.forEach(slice => {
-        let linkedSlices = getLinkedSlicesFromSubscriberLayer(slice.formData.subscriber_layers);
-        slice.formData.linked_slice = linkedSlices ? linkedSlices : slice.formData.linked_slice;
-        slice.formData.actions = ['APPLY_FILTER'];
+       slice.formData.linked_slice = getLinkedSlicesFromSubscriberLayer(slice.formData.subscriber_layers);
+       slice.formData.actions = getUniqueActionsForSlice(slice.formData.subscriber_layers);
+       slice.formData.useAsModal =  isSliceUsedAsModal(slice.formData.subscriber_layers);
```

Comment:
can we add this option at slice level internally before creating  pubsub map ,otherwise remove this options